### PR TITLE
fix: total fee calculation for estimate order

### LINKED ISSUE #6539: API/GraphQL: total fees in fee estimate wrong/badly formatted
# Problem encountered
When querying for a fee estimate on an order, the total fees turns up with multiple decimal points, and looks like a mangled concatenated string. (First seen by @MadalinaRaicu)

# Expected behaviour
A combined total of all estimated fees for the order

# System response
image (2)

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

### PR DESCRIPTION
**Pull request by jeremyletang**
close #6539

### CODE REVIEW

**Review by AnExsomnis — DISMISSED** [review]

File: datanode/gateway/graphql/resolvers.go [hunk 1]
```diff
@@ -883,7 +884,17 @@ func (r *myQueryResolver) EstimateOrder(
 	}
 
 	// calclate the fee total amount
-	ttf := resp.Fee.MakerFee + resp.Fee.InfrastructureFee + resp.Fee.LiquidityFee
+	var mfee, ifee, lfee num.Decimal
+	// errors doesn't matter here, they just gives us zero values anyway for the decimals
```
Comment (AnExsomnis) [inline datanode/gateway/graphql/resolvers.go:888]:
```suggestion
	// errors doesn't matter here, they just give us zero values anyway for the decimals
```

**Review by ValentinTrinque — DISMISSED** [review]

**Review by peterbarrow — DISMISSED** [review]

File: datanode/gateway/graphql/resolvers.go [hunk 2]
```diff
@@ -910,7 +921,7 @@ func (r *myQueryResolver) EstimateOrder(
 
 	return &OrderEstimate{
 		Fee:            &fee,
-		TotalFeeAmount: ttf,
+		TotalFeeAmount: (mfee.Add(ifee).Add(lfee)).String(),
```
Comment (ValentinTrinque) [inline datanode/gateway/graphql/resolvers.go:924]: Is a test possible ?

Comment (ze97286) [inline datanode/gateway/graphql/resolvers.go:924]: decimal.Sum(mfee, ifee, lfee)?

Comment (jeremyletang) [reply datanode/gateway/graphql/resolvers.go]: it's a pain to be fair, this API is calling to others APi and aggregate stuff out of it. For just a addition that sounds overdone.

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**